knpu/basics.rst

```diff
@@ -8,250 +7,236 @@ easy to learn, powerful and a joy to work with.
-        'products' => array(
-            'Tuxedo',
-            'Bow tie',
-            'Black Boxers',
-        ),
-    ));
-
-.. note::
-
-    If you're curious how you actually setup Twig, check out the code download
-    and see the `Twig Installation`_ documentation.
+Let's start by rendering a `homepage.twig` file and once again passing it
+`pageTitle` and `products` variables:
+
+```php
+// index.php
+// ... code that sets up Twig, and says to look for templates in template/
+
+echo $twig->render('homepage.twig', array(
+    'pageTitle' => 'Welcome to Penguins R Us!',
+    'products' => array(
+        'Tuxedo',
+        'Bow tie',
+        'Black Boxers',
+    ),
+));
+```
+
+> If you're curious how you actually setup Twig, check out the code download
+> and see the [Twig Installation][installation] documentation.
 
 If you're a frontend developer, then you don't need to worry about this step:
 all you need to know is where a Twig template is located and what variables
 you have access to.
 
-Your first Twig Template
-------------------------
+## Your first Twig Template
 
-In our project, Twig is looking for the template files in a ``templates/``
-directory, so let's create our ``homepage.twig`` there!
+In our project, Twig is looking for the template files in a `templates/`
+directory, so let's create our `homepage.twig` there!
 
 Just like in PHP, you can write anything and it'll just be displayed as HTML
 on the page:
 
-.. code-block:: html
+```html
+<!-- templates/homepage.twig -->
+Hello Twig Viewers!
+```
 
-    <!-- templates/homepage.twig -->
-    Hello Twig Viewers!
-
-To see this amazing message, go to the ``index.php`` file in your browser.
-This works because we made the ``index.php`` file render the ``homepage.twig``
+To see this amazing message, go to the `index.php` file in your browser.
+This works because we made the `index.php` file render the `homepage.twig`
 template. Whenever you're creating or editing a page, you'll need to figure
 out which Twig template is being used for that page. There's no exact science
 to this and it depends on how your application is built.
 
-Rendering a Variable
---------------------
+## Rendering a Variable
 
-Remember that we're passing a ``pageTitle`` variable to our template. To render
+Remember that we're passing a `pageTitle` variable to our template. To render
 it, write two opening curly braces, the name of the variable without a dollar
 sign, then two closing curly braces:
 
-.. code-block:: html+jinja
-
-    <!-- templates/homepage.twig -->
-    <h1>{{ pageTitle }}</h1>
+```html+jinja
+<!-- templates/homepage.twig -->
+<h1>{{ pageTitle }}</h1>
+```
 
 When we refresh the page, it works! We've just written our first line of Twig!
 Whenever you want to print something, just open Twig with two curly braces,
 write the variable name, then close Twig. We'll get fancier in a little while
 with some things called :ref:`functions<twig-functions>` and :ref:`filters<twig-filters>`,
```

Comment:
Yes, we *should* convert `:ref:`, it's a specific rst syntax only. And I already fixed it in further commits. I suppose you're reviewing a bit deprecated diff.